### PR TITLE
Add `exists` & `contains` methods for `JsResult`

### PR DESCRIPTION
**Pull request by aaabramov**
# Pull Request Checklist

* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Typesafe CLA](https://www.typesafe.com/contribute/cla)?
* [x] Have you [squashed your commits](https://www.playframework.com/documentation/latest/WorkingWithGit#Squashing-commits)?
* [x] Have you added copyright headers to new files?
* [x] Have you updated the documentation?
* [x] Have you added tests for any changed functionality?

### CODE REVIEW

**Comment by aaabramov** [comment]
Any suggestions how to properly add new methods to `play-json`?
```
[error] * method contains(java.lang.Object)Boolean in trait play.api.libs.json.JsResult is present only in current version
[error]   filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.libs.json.JsResult.contains")
[error] * method exists(scala.Function1)Boolean in trait play.api.libs.json.JsResult is present only in current version
[error]   filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.libs.json.JsResult.exists")
[info]play-json: found 2 potential binary incompatibilities while checking against com.typesafe.play:play-json_2.11:2.6.0  (filtered 17)
[error] * method contains(java.lang.Object)Boolean in trait play.api.libs.json.JsResult is present only in current version
[error]   filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.libs.json.JsResult.contains")
[error] * method exists(scala.Function1)Boolean in trait play.api.libs.json.JsResult is present only in current version
[error]   filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.libs.json.JsResult.exists")
```

**Comment by aaabramov** [comment]
@dwijnand could you help with this?

**Comment by renatocaval** [comment]
@aaabramov, you need to add the suggested filters on this part of the build configuration.

https://github.com/playframework/play-json/blob/master/build.sbt#L123-L144

It's safe to add the exclusions because `JsResult` is a sealed trait and therefore we know the existing implementations.

File: play-json/shared/src/main/scala/play/api/libs/json/JsResult.scala [hunk 1]
```diff
@@ -207,6 +207,12 @@ sealed trait JsResult[+A] { self =>
     def withFilter(q: A => Boolean) = new WithFilter(a => p(a) && q(a))
   }
 
+  /** If this result is successful than checks for presence for '''elem''', otherwise return '''false''' */
+  def contains[AA >: A](elem: AA): Boolean = asOpt.contains(elem)
```
Comment (cchantep) [inline play-json/shared/src/main/scala/play/api/libs/json/JsResult.scala]: Rather move implementation in the subtypes, as for the other functions.

File: play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala [hunk 2]
```diff
@@ -48,4 +48,37 @@ class JsResultSpec extends WordSpec with MustMatchers {
       JsError("err").recoverWith { _ => succ } mustEqual succ
     }
   }
+
+  "JsResult" should {
+
+    "return true for JsSuccess(x)#contains(x)" in {
+      val x = 10
```
Comment (cchantep) [inline play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala]: Do not need such trivial `val`

**Comment by aaabramov** [comment]
@cchantep, what about adding `forall` with `Option` semantics?

File: play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala (hunk 2 above)
Comment (aaabramov) [reply play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala]: @cchantep I don't agree with you. I think consistent test values and absence of magic numbers is good for tests.
If you would like to, I can rewrite tests with table driven cases.
Moreover, you can do it yourself :)

File: play-json/shared/src/main/scala/play/api/libs/json/JsResult.scala (hunk 1 above)
Comment (aaabramov) [reply play-json/shared/src/main/scala/play/api/libs/json/JsResult.scala]: Done.

File: play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala (hunk 2 above)
Comment (cchantep) [reply play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala]: There I see no value to declared such expected result as `val` as it's pure/simple/literal.

Comment (aaabramov) [reply play-json/shared/src/test/scala/play/api/libs/json/JsResultSpec.scala]: Do you think it is better to have
`assert(JsSuccess(10).contains(10))`?

**Comment by aaabramov** [comment]
@cchantep done.

**Comment by cchantep** [comment]
Thx, either add `forall` there or in a separate PR.

**Comment by aaabramov** [comment]
@cchantep `forall` added in scope of https://github.com/playframework/play-json/pull/184

**Pull request merged**